srv/adapter/stream.ts

```diff
@@ -47,12 +47,38 @@ export async function* needleToSSE(needleStream: NodeJS.ReadableStream) {
       throw new Error(`${error.message}: ${errorPayload}`)
     }
 
-    const newMsgs = (msgBuffer + chunks.join('')).split('\n\n')
+    const newMsgs = (msgBuffer + chunks.join('')).split(/\r?\n\r?\n/)
+
     chunks = []
     msgBuffer = newMsgs.pop() || ''
 
     for (const msg of newMsgs) {
-      yield msg
+      yield parseEvent(msg)
     }
   }
 }
+
+export type ServerSentEvent = { id?: string; type?: string; data: string }
+
+function parseEvent(msg: string) {
+  const buffer: ServerSentEvent = { data: '' }
+  return msg.split(/\r?\n/).reduce((event, line) => {
+    const sep = line.indexOf(':')
+    const field = sep === -1 ? line : line.slice(0, sep)
+    const value = sep === -1 ? '' : line.slice(sep + 1)
```

Comment:
Should handle:
- Empty lines (ignored)
- Lines beginning with colon ("comments", ignored)
- Lines with content, but no colon (content becomes `type`, empty `data`)
- Lines with content on both sides of colon (standard message with both `type` and `data`)